rcfiles/jointLklDM.rc

```diff
@@ -35,14 +35,24 @@ jointLklDM.plotsDir: plots
 # Comment out if you don't want to save or read the histograms but create them on the fly
 # (that is slower but safer because you make sure you use the proper histograms and not
 #  some pre-existing version created for a different set of IRFs or samples)
-jointLklDM.dNdEpSignalDir: DM/dNdEpSignal
+# Comment out for the channel 'branon' for now!!!
+# Comment out for the channel 'custom' always!!!
+#jointLklDM.dNdEpSignalDir: DM/dNdEpSignal
 
 ######################
 ANALYSIS CONFIGURATION
 ######################
 
-# channel to be analyzed (bb, tautau, mumu, WW, gammagamma, pi0pi0 or pi0gamma)
-jointLklDM.Channel: bb
+# channel to be analyzed (bb, tautau, mumu, WW, gammagamma, pi0pi0, pi0gamma, branon or custom)
+jointLklDM.Channel: custom
+
+# ONLY used if the channel 'custom' is selected. Otherwise, the lists will be ignored!
+# ChannelList contains a set of channels. 
+# Combinations of bb, tautau, mumu and WW are valid!
+# BRList contains the corresponding branching ratios.
+# The length of ChannelList and BRList have to be equal and the sum of the  BRList have to be1. 
+jointLklDM.ChannelList: bb tautau mumu WW
+jointLklDM.BRList: 1 0 0 0 
```

Comment:
Thanks for your suggestion. Having these strings is more user-friendly and less error-prone. I already adopted the format and modified jointLklDM to decode these strings. 
I will try to tackle the issue with SetdNdESignalFunction in the next days!

Reply:
I spent some time to get familiar with the `Iact1dUnbinned::SetdNdESignalFunction`/ `Iact1dUnbinned::AdddNdESignalFunction` and the structure of calling these function from the jointLklCM.C script. I was thinking about a strategy to add the multichannel feature, but is a bit tricky to find a good solution. A good and clean approach (in my point of view) to solve this problem is to move some parts of the jointLklDM.C into the Iact1dUnbinned class. Since this will change the structure and the arguments of the function, I want to suggest it to you before I start implementing it. Maybe I'm missing some important points, which you are aware of.

My suggestion:
- We are calling a new `Iact1dUnbinned::SetdNdESignalFunction` with the arguments (pointer to a list of channels, pointer to a list of the corresponding branching ratios, mass value "mdm").
This call would be also valid when the lists only contain one element (100% BR into single channel)
- We will move the channel discrimination in the Iact1dUnbinned class and call the `SetdNdESignalFunction` from there. Therefore the user, who is likely interacting on rc configuration file- and script-level, don't have to worry about setting up the right `SetdNdESignalFunction` with the right parameters for each channel. The user only have to call one function with a channel and branching ratio list plus the mass value "mdm" and it will automatically set up the right dN/dE histogram for them.

I think is a clean and intuitive way to tackle this problem. Please let me know if you have any doubts.

Reply:
The problem with this approach is that Iact1dUnbinned should be agnostic about the physics behind the signal spectra it's looking for. 
What is the difference between your proposal and doing exactly the same thing in the script?
My idea is that all scripts should become executables at some point and user should not be allowed to change them. Users should only touch the rc files.
Is there any fundamental reason why the interpretation of the setting up of the dNdE histogram cannot happen in the script?
If there really is, I am open to include it in the class, but we need a new class which inherits from Iact1dUnbinned and contains the DM-related part. I could take care of it, because some parts of the current Iact1dUnbinned class should go to the new one. But let's see first if it's really necessary

Reply:
Oh I see! I forgot the point that gLike is a general tool and not specific for DM searches! It makes totally sense to have every DM relative things in the DM script and not in the general Unbinned lkl class. My concern was only based on code readability and compactness.

Reply:
This is solved with the latest commit.